validate/validate.py

```diff
@@ -60,22 +51,19 @@ def print_success(path:Path, file_type:str):
     ''' Prints success message '''
     print(f'Input: {path} is valid {file_type}')
 
-def detect_file_type_and_extension(path:Path):
-    ''' File type and extension detection '''
+def detect_extension(path:Path):
+    ''' File extension detection '''
     # Starting from the end, build up extension with each '.' separated part and try matching
     # resulting extension
     full_extension = ''
     for suffix in path.suffixes[::-1]:
         full_extension = suffix + full_extension
-        extension_type = check_extension(full_extension)
-
-        if extension_type != UNKNOWN_FILE_TYPE:
-            return extension_type, full_extension
-
+        if full_extension not in COMPRESSION_TYPES:
```

Comment:
Let's keep the explicit extension check here; we'll eventually expand the validation tools/file types and extensions so the explicit check should be more robust/maintainable

Reply:
I think I'll have to just change it back to when it returned both the file type and the extension then, if that's okay. 
Do you have a preference on whether the check extension and check file type functions are separated? I can't think of a way to keep them separated at the moment without the compression extension list. Unless you have suggestions. 

---
It seems like it the way it was would check to see for every `.` whether the extension was in the list.

originally:
each found suffix (specified by `.`):
     loop through list of file types and check to see if the full extension exists and return the file type. 
     
so if the suffix was something like .gz then it would be doing that loop twice in the case of `.vcf.gz`. since .gz (by itself) is not in the list. and it has to call that check_extension (which is actually detect_filetype) for every suffix until it finds the full extension in the list or runs out of suffixes. 

I don't think this would have a big impact on performance since I really can't imagine we'd have significantly more file type and compression permutations. worst case it isn't one we support (eg. compressed bams or something) and somehow it's a bunch of `.`'s in the filename and there are tons of files.

I could be wrong though.